lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -154,9 +154,18 @@ Parser.prototype.onFrameBody = function onFrameBody(header, buffer, callback) {
   else if (header.type === frameType.X_FORWARDED_FOR)
     this.onXForwardedFrame(header, buffer, callback);
   else
-    callback(null, { type: 'unknown: ' + header.type });
+    this.onUnknownFrame(header, buffer, callback);
 };
 
+Parser.prototype.onUnknownFrame = function onUnknownFrame(header, buffer, callback) {
```

Comment:
It seems another frames is valid in-between of HEADERS.
h2spec 5.5. Extending HTTP/2 just checks:
    ✓ Sends an unknown extension frame
    ✓ Sends an unknown extension frame in the middle of a header block
So both checks are passed and none of "in the middle" checks in list.